torch/csrc/Module.cpp

```diff
@@ -688,6 +690,40 @@ PyObject* THPModule_isEnabledXNNPACK(PyObject* _unused, PyObject* noargs) {
     Py_RETURN_FALSE;
 }
 
+PyObject* THPModule_willEngineExecuteNode(PyObject* _unused, PyObject* arg) {
+  HANDLE_TH_ERRORS
+  bool isTHPFunction = THPFunction_Check(arg);
+  bool isTHPCppFunction = torch::autograd::THPCppFunction_Check(arg);
+  THPUtils_assert(
+      isTHPFunction || isTHPCppFunction,
+      "_will_engine_execute_node expects an grad_fn, "
+      "but got %s",
+      THPUtils_typename(arg));
+  const auto exec_info = torch::autograd::get_current_graph_task_exec_info();
+  THPUtils_assert(
+      exec_info,
+      "_get_should_execute_nodes should only be called during the backward pass");
+  THPUtils_assert(
+      !exec_info->empty(),
+      "_get_should_execute_nodes should only be used with .grad() or .backward() "
```

Comment:
Why is that?
Even in the case of `backward()`, we need to know how many gradients will be computed for this function and if that number is >1 that means this function will be executed.

Reply:
We don't want to erroneously return True when there is a Tensor that is not part the graph, so we'd have to know what nodes will be traversed, when we eventually traverse the graph (I'm not sure if we have this information, but I can double check). If it's not already there, adding it seems to be additional overhead that we might want to avoid.

Reply:
If the Tensor is not part of the graph, then this Node will have 0 required gradients (or more likely no entry in the required gradients table). Meaning that no gradient will be computed for it.
This is stored in the `dependencies_` field of the graph task btw.

Reply:
Thanks for the pointer, updated!